subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -55,6 +55,15 @@ eclipse {
 
 Alternatively, you can adjust or remove classpath attributes in the `eclipse.classpath.file.whenMerged { }` block.
 
+=== Deprecations
+
+[[file_collection_to_classpath]]
+==== Converting files to a classpath where paths contain file separator
+Gradle 7.5+ until 8.0 will emit a deprecation warning when using
+`@link:{javadocPath}/javadoc/org/gradle/api/file/FileCollection.html#getAsPath--[FileCollection.getPath()]` for files with paths that contain a path separator. Allowing paths with the path separator can, in some cases, result in an uncontrolled failure of the build.
+
+In Gradle 8.0 and later this will be an error.
```

Comment:
```suggestion
Java has the concept of a path separator which is used to separate individual paths in a list of paths, for example in a classpath string.
The individual paths must not contain the path separator.
Consequently, using
`@link:{javadocPath}/javadoc/org/gradle/api/file/FileCollection.html#getAsPath--[FileCollection.getPath()]` for files with paths that contain a path separator has been deprecated.
Using a file collection with paths which contain a path separator may lead to incorrect builds, since Gradle doesn't find the files as inputs, or even to build failures when the path containing the path separator is illegal on the operating system.
```

A few comments:
* In documentation, we try to use one-line-per-sentence style, so it is easier to do review comments.
* You don't need to take about the lifecycle of deprecations, since that is already something we document in our user manual.
* It is good to give some context what the feature is about.

I left a suggestion how to make it better. Please take a look and tell me what you think.

Reply:
I think it's good and gives more context to the user.

I am wondering if we should point out that this will be an error in 8.0, so users are reminded about that. What do you think?

Reply:
I don't think it adds much value, though you can if you want to. I looked through the upgrading notes and some deprecations have something about 8.0, while others don't. Given that the default is to fail in Gradle 8.0, I am not sure if pointing that out gives more value.

Reply:
I added `, and it will be an error in Gradle 8.0 and later`.